daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -135,9 +135,33 @@ object SExpr {
     }
   }
 
+  // A fully saturated builtin application
+  final case class SEAppB(builtin: SBuiltin, args: Array[SExpr])
+      extends SExpr
+      with SomeArrayEquals {
+    if (args.size != builtin.arity) {
+      throw SErrorCrash(s"SEAppB: arg.size != builtin.arity")
+    }
+    def execute(machine: Machine): Unit = {
+      val arity = builtin.arity
+      val actuals = new util.ArrayList[SValue](arity)
+      machine.pushKont(KBuiltin(builtin, actuals))
+      // Start evaluating the arguments.
+      var i = 1
```

Comment:
I think we have this loop in multiple places now. Should we factor it out into its own function to simplify future refactorings?

Reply:
ok, I'll do this...

Reply:
done